frontend/src/modals/new-document/NewDocumentModal.tsx

```diff
@@ -39,147 +61,184 @@ const createDocumentMutation = gql`
+          const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message };
+          dispatch(showToastEvent);
+        }
+      }
+    }
+  }, [createDocumentLoading, createDocumentData, createDocumentError, dispatch]);
+
+  const onCreateDocument = () => {
+    dispatch({ type: 'CREATE_DOCUMENT' } as CreateDocumentEvent);
+    const input = {
+      id: uuid(),
+      editingContextId,
+      name,
+      stereotypeDescriptionId: selectedStereotypeDescriptionId,
     };
-    createDocument({ variables });
+    createDocument({ variables: { input } });
   };
+
   useEffect(() => {
-    if (!createDocumentLoading && !createDocumentError && createDocumentData?.createDocument) {
-      const { createDocument } = createDocumentData;
-      if (createDocument.__typename === 'CreateDocumentSuccessPayload') {
-        const { id } = createDocument.document;
-        onDocumentCreated(id);
-      } else if (createDocument.__typename === 'ErrorPayload') {
-        setState((prevState) => {
-          const newState = { ...prevState };
-          newState.message = createDocument.message;
-          newState.isValid = false;
-          return newState;
-        });
-      }
+    if (newDocumentModal === 'success') {
+      onDocumentCreated();
     }
-  }, [createDocumentLoading, createDocumentData, createDocumentError, onDocumentCreated]);
+  }, [newDocumentModal, onDocumentCreated]);
 
   return (
-    <Modal title="Create a new model" onClose={onClose}>
-      <Form onSubmit={createNewDocument}>
-        <Label value="Name">
-          <Textfield
-            name="name"
-            placeholder="Enter the model name"
-            value={name}
-            onChange={onNewName}
-            autoFocus
-            data-testid="name"
-          />
-        </Label>
-        <Label value="Model type">
-          <Select
-            name="stereotype"
-            value={selectedStereotypeDescriptionId}
-            options={stereotypeDescriptions}
-            onChange={onNewSelectedStereotypeDescription}
-            data-testid="stereotype"
-          />
-        </Label>
-        <Buttons>
-          <ActionButton type="submit" disabled={!isValid} label="Create" data-testid="create-document" />
-        </Buttons>
-      </Form>
-    </Modal>
+    <>
+      <Dialog open={true} onClose={onClose} aria-labelledby="dialog-title" maxWidth="xs" fullWidth>
+        <DialogTitle id="dialog-title">Create a new model</DialogTitle>
+        <DialogContent>
+          <FormGroup>
+            <TextField
+              error={nameIsInvalid}
+              helperText={nameMessage}
+              label="Name"
+              name="name"
+              value={name}
+              placeholder="Enter the name of the document"
+              data-testid="name"
+              autoFocus={true}
+              onChange={onNameChange}
+              disabled={newDocumentModal === 'loading' || newDocumentModal === 'creatingDocument'}
+            />
+            <Select
```

Comment:
The previous UI had a label "Model type" for this Select which seems to be lost now.